<?xml version='1.1' encoding='UTF-8'?>
<Tag plugin="workflow-support@3.2">
  <node class="cps.n.StepAtomNode" plugin="workflow-cps@2.63">
    <parentIds>
      <string>26</string>
    </parentIds>
    <id>27</id>
    <descriptorId>org.jenkinsci.plugins.workflow.steps.durable_task.ShellStep</descriptorId>
  </node>
  <actions>
    <cps.a.ArgumentsActionImpl plugin="workflow-cps@2.63">
      <arguments>
        <entry>
          <string>script</string>
          <string>
					set +x
					file=&quot;$(pwd)/ABC.exe&quot;
					pwd;
					
				echo -e &quot;

**************** This is a Deployed JOB Testing for $file******************* &quot;
				if [[ -f &quot;$file&quot; ]]; then 
				count=1		
					num1=$((1 + RANDOM % 100))
					num2=$((1 + RANDOM % 1000))
					num3=$((1 + RANDOM % 2000))
					
						echo -e &quot;
-------------------------- TEST_CASE ($count) --------------------------&quot;
						$file -v -i &lt;&lt;&lt;&quot;$num1 $num2 num3&quot;
					echo $?
					if [ $? -eq 0 ]; then
						echo -e &quot;
      RESULT -&gt; TEST_CASE (${count}): SUCCESS&quot; 
					else  
						echo -e &quot;
      RESULT -&gt; TEST_CASE (${count}): FAILED&quot;
						exit 1
					fi
				
				else 
					 echo -e &quot;**************** ERROR *********************
&quot;
					  echo &quot;Test Build Failure - ABC.exe notfound $file &quot;
					  echo -e &quot;********************************************
&quot;
					  exit 1
				fi
				</string>
        </entry>
      </arguments>
      <isUnmodifiedBySanitization>true</isUnmodifiedBySanitization>
    </cps.a.ArgumentsActionImpl>
    <wf.a.TimingAction plugin="workflow-api@2.33">
      <startTime>1550499409362</startTime>
    </wf.a.TimingAction>
    <s.a.LogStorageAction/>
    <wf.a.ErrorAction plugin="workflow-api@2.33">
      <error class="hudson.AbortException">
        <detailMessage>script returned exit code 34</detailMessage>
        <stackTrace>
          <trace>org.jenkinsci.plugins.workflow.steps.durable_task.DurableTaskStep$Execution.handleExit(DurableTaskStep.java:561)</trace>
          <trace>org.jenkinsci.plugins.workflow.steps.durable_task.DurableTaskStep$Execution.check(DurableTaskStep.java:507)</trace>
          <trace>org.jenkinsci.plugins.workflow.steps.durable_task.DurableTaskStep$Execution.run(DurableTaskStep.java:453)</trace>
          <trace>java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)</trace>
          <trace>java.util.concurrent.FutureTask.run(FutureTask.java:266)</trace>
          <trace>java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)</trace>
          <trace>java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)</trace>
          <trace>java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)</trace>
          <trace>java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)</trace>
          <trace>java.lang.Thread.run(Thread.java:748)</trace>
        </stackTrace>
        <suppressedExceptions class="java.util.Collections$UnmodifiableRandomAccessList" resolves-to="java.util.Collections$UnmodifiableList">
          <c class="list"/>
          <list reference="../c"/>
        </suppressedExceptions>
      </error>
    </wf.a.ErrorAction>
  </actions>
</Tag>